ocaml/xapi-guard/src/main.ml

```diff
@@ -22,6 +22,11 @@ let ret v = v >>= Lwt.return_ok |> Rpc_lwt.T.put
 
 let sockets = Hashtbl.create 127
 
+let log_fds () =
+  let fds = Sys.readdir "/proc/self/fd" |> Array.length in
```

Comment:
You can use `Lwt_unix.files_of_directory` and then count the number of entries in the Lwt stream if we want to avoid blocking here (Sys.readdir would be a blocking call that blocks all Lwt promises)

Reply:
```

Reply:
(there are a few more places in varstored_guard that use blocking calls like Sys.file_exists, don't worry about those I'll try to fix those in my PR)

Reply:
I think you meant
```ocaml
let stream_length s = Lwt_stream.fold (fun _ acc -> acc + 1) s 0 in
let fds = Lwt_unix.files_of_directory "/proc/self/fd" |> stream_length
```